packages/webview_flutter/webview_flutter_android/lib/webview_android_widget.dart

```diff
@@ -0,0 +1,632 @@
+      updateSettings(webSettings);
+    }
+
+    final String? userAgent = creationParams.userAgent;
+    if (userAgent != null) {
+      webView.settings.setUserAgentString(userAgent);
+    }
+
+    final AutoMediaPlaybackPolicy autoMediaPlaybackPolicy =
+        creationParams.autoMediaPlaybackPolicy;
+    switch (autoMediaPlaybackPolicy) {
+      case AutoMediaPlaybackPolicy.always_allow:
+        webView.settings.setMediaPlaybackRequiresUserGesture(false);
+        break;
+      default:
+        webView.settings.setMediaPlaybackRequiresUserGesture(true);
+    }
+
+    addJavascriptChannels(creationParams.javascriptChannelNames);
+  }
+
+  Future<void> _trySetHasProgressTracking(bool? hasProgressTracking) {
+    if (hasProgressTracking == true) {
+      webChromeClient._onProgress = callbacksHandler.onProgress;
+    } else if (hasProgressTracking == false) {
+      webChromeClient._onProgress = null;
+    }
+
+    return Future<void>.sync(() => null);
+  }
+
+  Future<void> _trySetHasNavigationDelegate(bool? hasNavigationDelegate) {
+    if (hasNavigationDelegate == null) return Future<void>.sync(() => null);
+
+    downloadListener._onNavigationRequest =
+        callbacksHandler.onNavigationRequest;
+    if (hasNavigationDelegate) {
+      _webViewClient = WebViewAndroidWebViewClient.handlesNavigation(
+        onPageStartedCallback: callbacksHandler.onPageStarted,
+        onPageFinishedCallback: callbacksHandler.onPageFinished,
+        onWebResourceErrorCallback: callbacksHandler.onWebResourceError,
+        loadUrl: loadUrl,
+        onNavigationRequestCallback: callbacksHandler.onNavigationRequest,
+      );
+    } else {
+      _webViewClient = WebViewAndroidWebViewClient(
+        onPageStartedCallback: callbacksHandler.onPageStarted,
+        onPageFinishedCallback: callbacksHandler.onPageFinished,
+        onWebResourceErrorCallback: callbacksHandler.onWebResourceError,
+      );
+    }
+    return webView.setWebViewClient(_webViewClient);
+  }
+
+  Future<void> _trySetJavaScriptMode(JavascriptMode? mode) async {
+    if (mode == null) return Future<void>.sync(() => null);
+
+    switch (mode) {
+      case JavascriptMode.disabled:
+        return webView.settings.setJavaScriptEnabled(false);
+      case JavascriptMode.unrestricted:
+        return webView.settings.setJavaScriptEnabled(true);
+    }
+  }
+
+  Future<void> _trySetDebuggingEnabled(bool? debuggingEnabled) async {
+    if (debuggingEnabled == null) return Future<void>.sync(() => null);
+    return webViewProxy.setWebContentsDebuggingEnabled(
+      debuggingEnabled,
+    );
+  }
+
+  Future<void> _trySetUserAgent(WebSetting<String?> userAgent) {
+    if (userAgent.isPresent && userAgent.value != null) {
+      return webView.settings.setUserAgentString(userAgent.value!);
+    }
+
+    return webView.settings.setUserAgentString('');
+  }
+
+  Future<void> _trySetZoomEnabled(bool? zoomEnabled) async {
+    if (zoomEnabled == null) return Future<void>.sync(() => null);
```

Comment:
This inline condition+statement reminds me: since you're rewriting ~all of the Dart, how hard would it be to flip the analysis options of this package to the new options? It would substantially cut down on later churn.

(To do just this one, you'd move packages/webview_flutter/analysis_option.yaml into all of the subpackages except for this one.)

Reply:
Done. I updated the analysis options for this plugin